### PR TITLE
Add `HfFileSystem.url` method

### PR DESCRIPTION
**Pull request by mariosasko**
Adds a `url` method to the `HfFileSystem` to simplify converting HF paths to HTTP URLs, which should be useful when working with libs that support HTTP URLs but not `fsspec` paths as input/output (e.g., `webdataset`, `polars`, etc.).

PS: The `url` method is not part of the official `fsspec` specification, but popular filesystem implementations such as `gcsfs` and `s3fs` also have it

### CODE REVIEW

File: src/huggingface_hub/hf_file_system.py [hunk 1]
```diff
@@ -577,6 +577,19 @@ def isfile(self, path):
...
+        """Get the HTTP URL of the given path"""
+        resolved_path = self.resolve_path(path)
+        url = hf_hub_url(
+            resolved_path.repo_id,
+            resolved_path.path_in_repo,
+            repo_type=resolved_path.repo_type,
+            revision=resolved_path.revision,
+            endpoint=self.endpoint,
+        )
+        url = url.replace("/resolve/", "/tree/", 1) if self.isdir(path) else url
```
Comment (Wauplin) [inline src/huggingface_hub/hf_file_system.py:590]:
```suggestion
        if self.isdir(path) 
            url = url.replace("/resolve/", "/tree/", 1)
```

(nit)

File: src/huggingface_hub/hf_file_system.py [hunk 2]
```diff
@@ -577,6 +577,19 @@ def isfile(self, path):
         except:  # noqa: E722
             return False
 
+    def url(self, path):
```
Comment (Wauplin) [inline src/huggingface_hub/hf_file_system.py:580]:
```suggestion
    def url(self, path: str) -> str:
```

File: src/huggingface_hub/hf_file_system.py [hunk 3]
```diff
@@ -653,6 +666,9 @@ def _upload_chunk(self, final: bool = False) -> None:
                 path=self.resolved_path.unresolve(),
             )
 
+    def url(self):
```
Comment (Wauplin) [inline src/huggingface_hub/hf_file_system.py:669]:
```suggestion
    def url(self) -> str:
```

File: src/huggingface_hub/hf_file_system.py [hunk 4]
```diff
@@ -740,6 +756,9 @@ def read(self, length: int = -1):
         self.loc += len(out)
         return out
 
+    def url(self):
```
Comment (Wauplin) [inline src/huggingface_hub/hf_file_system.py:759]:
```suggestion
    def url(self) -> str:
```

**Review by Wauplin — APPROVED** [review]
Thanks for the addition! Left a few nits but looks good to me :)

**Review by lhoestq — APPROVED** [review]
Nice thanks !

**Review by Wauplin — APPROVED** [review]
Thanks for making the changes!

**Pull request merged**